src/components/Quiz.js

```diff
@@ -18,19 +18,63 @@ const Quiz = () => {
     setQuestions(quizData);
   }, [quizName]);
 
-  const handleAnswerSelect = e => {
-    const selectedAnswer = { [e.target.name]: Number(e.target.value) };
+  const handleCheckboxAnswers = (e, answers) => {
+    const questionIds = Object.keys(answers || {});
+    console.log('questionIds', questionIds);
+    const currentQuestionId =
+      questionIds &&
+      questionIds.filter(id => Number(id) === Number(e.target.name))[0];
+
+    let checkedAnswer;
+    if (currentQuestionId === e.target.name) {
+      const previousSelectedAnswers = answers[currentQuestionId];
+      const isPreviouslySelected = previousSelectedAnswers.filter(
+        answer => answer === Number(e.target.value),
+      );
+      if (isPreviouslySelected.length) {
+        const removePreviousAnswer = previousSelectedAnswers.filter(
+          answer => answer !== Number(e.target.value),
+        );
+        checkedAnswer = {
+          [e.target.name]: removePreviousAnswer,
+        };
+      } else {
+        checkedAnswer = {
+          [e.target.name]: [...previousSelectedAnswers, Number(e.target.value)],
+        };
+      }
+    } else {
+      checkedAnswer = { [e.target.name]: [Number(e.target.value)] };
+    }
+    return checkedAnswer;
+  };
+
+  const handleAnswerSelect = (e, type) => {
+    const answers = userAnswers;
+    let selectedAnswer;
+    if (Object.keys(answers).length && type === 'checkbox') {
+      selectedAnswer = handleCheckboxAnswers(e, answers);
+    } else {
+      selectedAnswer = { [e.target.name]: [Number(e.target.value)] };
+    }
     setUserAnswers({ ...userAnswers, ...selectedAnswer });
   };
 
   const checkUserAnswers = e => {
     e.preventDefault();
     const totalScore = questions.reduce((point, question) => {
-      const selectedAnswerId = userAnswers[question.id];
-      const selectedAnswerObj = question.answers.find(
-        answer => answer.id === selectedAnswerId,
-      );
-      return selectedAnswerObj.isCorrect ? point + 1 : point;
+      const selectedAnswer = userAnswers[question.id];
+      const correctAnswer = question.answers
+        .filter(ans => ans.isCorrect)
+        .map(correctAns => correctAns.id);
+
+      let isRight = false;
+      if (selectedAnswer.length === correctAnswer.length) {
+        isRight = selectedAnswer.every(
+          answer => correctAnswer.indexOf(answer) !== -1,
+        );
+      }
```

Comment:
```suggestion
      const isRight = question.answers.every(answer =>
        // predicate
      );
```
There is a way to perform all this logic in just one `every` method call. Lets see if we can figure it out? Give it a go — I can provide the answer if needed

Reply:
@bonboh don't provide the answer! Give @abdulkadiret just a hint! He has to figure out by himself (Laughing, joking :)

Reply:
Haha. A hint is that `answer.isCorrect` and `includes` are some of the bits of my code 😄